crates/core/src/runloops/mod.rs

```diff
@@ -226,51 +233,51 @@ pub fn start_clock_runloop(mut slot_time: u64) -> (Receiver<ClockEvent>, Sender<
 }
 
 fn start_geyser_runloop(
+    plugin_config_paths: Vec<PathBuf>,
     plugin_manager_commands_rx: Receiver<PluginManagerCommand>,
     subgraph_commands_tx: Sender<SubgraphCommand>,
     simnet_events_tx: Sender<SimnetEvent>,
     geyser_events_rx: Receiver<GeyserEvent>,
 ) -> Result<JoinHandle<Result<(), String>>, String> {
     let handle = hiro_system_kit::thread_named("Geyser Plugins Handler").spawn(move || {
-        let mut plugin_manager = vec![];
+        let mut plugin_manager = GeyserPluginManager::new();
+        let mut surfpool_plugin = None;
+
+        for plugin_config_path in plugin_config_paths.into_iter() {
+            let plugin_manifest_location = FileLocation::from_path(plugin_config_path);
+            let contents = plugin_manifest_location.read_content_as_utf8()?;
+            let result: serde_json::Value = json5::from_str(&contents).unwrap();
```

Comment:
Could `.map_err(...)?;` here? Seems quite likely that a user accidentally points to a non-json file, and a helpful error would be great for that